portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -72,18 +73,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
         // If video id doesn't exist in database, convert to video entity and add to database.
         if(queryVideoEntity == null){
             Video video;
-            try {
-                video = AUX.VideoIdToObject(videoId);
-            } catch (IOException e) {
-                sendErrorMessage(response, HttpServletResponse.SC_NOT_FOUND, "Unable to retrieve video.");
-                return;
-            } catch (IndexOutOfBoundsException e) {
-                sendErrorMessage(response, HttpServletResponse.SC_NOT_FOUND, "No such video found.");
-                return;
-            } catch (GeneralSecurityException e) {
-                sendErrorMessage(response, HttpServletResponse.SC_NOT_FOUND, "A security exception occurred.");
+            try{
+                video = AUX.VideoIdToObject(id);
+            }
+            catch (VideoException e){
+                sendErrorMessage(response, HttpServletResponse.SC_BAD_REQUEST, e.toString());
```

Comment:
e.toString() will output something like:
"java.io.FileNotFoundException: yourFile.txt (The system cannot find the file specified)"
which is not really something we want the user to see.
e.getMessage() will return the original message we passed in without any extra code information, so I suggest using that instead.